zstash/extract.py

```diff
@@ -457,8 +466,10 @@ def extractFiles(  # noqa: C901
 
             # Delete this tar if the corresponding command-line arg was used.
             if not keep_tars:
-                # FIXME: Argument 1 to "remove" has incompatible type "Optional[Any]"; expected "Union[str, bytes, _PathLike[str], _PathLike[bytes]]"mypy(error)
-                os.remove(tfname)  # type: ignore
+                if tfname:
+                    os.remove(tfname)
+                else:
+                    raise Exception("Invalid tfname={}".format(tfname))
```

Comment:
You should use raise ValueError(...) to be more explicit on what the exception is.

https://docs.python.org/3/library/exceptions.html#ValueError
https://stackoverflow.com/a/43605619